src/replication/replicator.rst

```diff
@@ -155,98 +203,192 @@ the following order:
+
+White shapes indicate internal API boundaries and point to how the
+replicator is structured internally. There are two stages in the
+processing: the first is where replication documents are parsed and
+become replication jobs, and the second is the scheduler.  The
+scheduler runs replication jobs, periodically stopping and starting
+some. Jobs posted via the ``_replicate`` endpoint bypass the first
+component and go straight to the scheduler.
+
+States descriptions
+-------------------
+
+Before explaining the details of each state, it is worth noticing that
+color and shape of each state in the diagram:
+
+`Green` vs `red` partitions states into "healthy" and "unhealthy",
+respectively. Unhealthy states indicate something has gone wrong and
+it might need user's attention.
+
+`Rectangle` vs `oval` separates "terminal" states from "non-terminal"
+ones. Terminal states are those which will not transition to other
+states any more. Informally, jobs in a terminal state will not be
+retried and don't consume memory or CPU resources.
+
+ * ``Initializing``: Indicates replicator has noticed the change
+   from the replication document. Jobs should transition quickly
+   through this state. Being stuck here for a while could mean there
+   is an internal error.
+
+ * ``Failed``: Replication document could not be processed and turned
+   into a valid replication job for the scheduler. This state is
+   terminal and requires user intervention to fix the problem. Typical
+   reasons for ending up in this state is a malformed document. For
+   example specifying an integer for a parameter which accepts a
+   boolean. Another reason could be specifying a duplicate
+   replication. A duplicate replication is a replication with
+   identical parameters but a different document ID.
+
+ * ``Error``: Replication document update could not be turned into a
+   replication job. Unlike the ``Failed`` state, this one is
+   temporary, and replicator will keep retrying periodically. There is
+   an exponential backoff applied in case of consecutive failures.
+   The main reason this state exists is to handle filtered
+   replications with custom user functions. Filter function content is
+   needed in order to calculate the replication ID. A replication job
+   could not be created until the function code is retrieved. Because
+   retrieval happens over the network, temporary failures have to be
+   handled.
+
+ * ``Running``: Replication job is running normally. This means,
+   there might be a change feed open, and if changes are noticed, they
+   would be processed and posted to the target. Job is still
+   considered ``Running`` even if its workers are currently not
+   streaming changes from source to target and are just waiting on the
+   change feed.
+
+ * ``Pending``: Replication job is not running and is waiting its
+   turn. This state is reached then more than ``replicator.max_jobs``
+   replication have been added to the scheduler. In that case
```

Comment:
This state is reached when replication jobs added to the scheduler exceeds replicator.maxjobs.